File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode reverseList(ListNode head) {
        if (head == null) {
            return head;
        }
        ListNode reverseHead = new ListNode(head.val, null);
        while (head.next != null) {
            head = head.next;
            reverseHead = new ListNode(head.val, reverseHead);
        }
        return reverseHead;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False
        dict = {}
        for l in s:
            if l in dict:
                dict[l] += 1
            else:
                dict[l] = 1
        for l in t:
            if l in dict:
                dict[l] -= 1
            else:
                return False
        for val in dict.values():
            if val != 0:
                return False
        return True